bench/state_transition/process_epoch.zig

```diff
@@ -0,0 +1,702 @@
+    defer std.process.argsFree(allocator, args);
+
+    const state_path = if (args.len > 1) args[1] else "bench/state_transition/state.ssz";
+
+    try stdout.print("Loading state from {s}...\n", .{state_path});
+
+    const state_file = try std.fs.cwd().openFile(state_path, .{});
+    defer state_file.close();
+    const state_bytes = try state_file.readToEndAlloc(allocator, 10_000_000_000);
+    defer allocator.free(state_bytes);
+
+    try stdout.print("State file loaded: {} bytes\n", .{state_bytes.len});
+
+    // Detect fork from state SSZ bytes
+    const chain_config = config.mainnet_chain_config;
+    const slot = config.slotFromStateBytes(state_bytes) orelse {
+        try stdout.print("Error: Could not read slot from state SSZ bytes\n", .{});
+        return error.InvalidStateBytes;
+    };
+    const detected_fork = config.forkSeqAtSlot(chain_config, slot);
+    try stdout.print("Detected fork: {s} (slot {})\n", .{ @tagName(detected_fork), slot });
+
+    // Dispatch to fork-specific loading
+    inline for (comptime std.enums.values(ForkSeq)) |fork| {
+        if (detected_fork == fork) {
+            return runBenchmark(fork, allocator, stdout, state_bytes, chain_config);
+        }
+    }
+}
+
+fn runBenchmark(
+    comptime fork: ForkSeq,
+    allocator: std.mem.Allocator,
+    stdout: anytype,
+    state_bytes: []const u8,
+    chain_config: config.ChainConfig,
+) !void {
+    const beacon_state = try loadState(fork, allocator, state_bytes);
+    try stdout.print("State deserialized: slot={}, validators={}\n", .{
+        beacon_state.slot(),
+        beacon_state.validators().items.len,
+    });
+
+    const beacon_config = try config.BeaconConfig.init(allocator, chain_config, beacon_state.genesisValidatorsRoot());
+
+    const pubkey_index_map = try PubkeyIndexMap.init(allocator);
+    const index_pubkey_cache = try allocator.create(state_transition.Index2PubkeyCache);
+    index_pubkey_cache.* = state_transition.Index2PubkeyCache.init(allocator);
+
+    try state_transition.syncPubkeys(beacon_state.validators().items, pubkey_index_map, index_pubkey_cache);
+
+    const immutable_data = state_transition.EpochCacheImmutableData{
+        .config = beacon_config,
+        .index_to_pubkey = index_pubkey_cache,
+        .pubkey_to_index = pubkey_index_map,
+    };
+
+    const cached_state = try CachedBeaconStateAllForks.createCachedBeaconState(allocator, beacon_state, immutable_data, .{
+        .skip_sync_committee_cache = !comptime fork.isPostAltair(),
+        .skip_sync_pubkeys = false,
+    });
+
+    try stdout.print("Cached state created at slot {}\n", .{cached_state.state.slot()});
+    try stdout.print("\nStarting process_epoch benchmarks for {s} fork...\n\n", .{@tagName(fork)});
+
+    var bench = zbench.Benchmark.init(allocator, .{ .iterations = 50 });
+    defer bench.deinit();
+
+    // All forks
+    try bench.addParam("justification_finalization", &ProcessJustificationAndFinalizationBench{
```

Comment:
```suggestion
    try bench.addParam("justification_finalization", &ProcessJustificationAndFinalizationBench{
```

imo it's clear to the reader that these benchmarks are ran for all forks (especially after seeing the other benchmarks below)

Same to the other comments for the rest of the scope